### ISSUE TITLE
Move page (and every link pointing to it) up the directory hierarchy? 

### ISSUE THREAD

**Issue by notevenaperson** [vimwiki/vimwiki#926]
After using vimwiki for a while I realized that organizing my pages in many subfolders takes more effort than an immaculate file tree is worth, and makes linking a more time-consuming process.

I know of the VimwikiRenameFile function, which renames a file and updates links to it across the whole wiki, but I can't find one to move a file to a different directory without breaking every link.

**Comment by tinmarino** [comment]
Seems like a bug :VimwikiRenameFile should work

**Comment by iandstanley** [comment]
I've really gotten into Vimwiki and it's great tools but I have hit an issue of too many files as well and I need to reorganise.

I have created a number of subdirectories and then tried :VimwikiRenameFile to migrate the file (I cant find any documentation to suggest another method) 

When I try to migrate a file (say **linux.wiki** to **linux/linux.wiki**) with the command **:VimwikiRenameFile linux/linux** 
i get an error 
**E488: trailing characters**

the same issue occurs with :VimwikiRenameLink 

I can't see a way of migrating content into a subdirectory without having either an error or manually updating links and probably missing loads

**Comment by tinmarino** [comment]
@notevenaperson Actually it wasn't a bug, my bad: VimwikiRenameFile did not support files in other dir
@iandstanley VimwikiRenameFile was not supposed to take argument: press enter and it ask you to input the file name

I considered both as feature request, I coded it and now in dev you can:
1. Move file to an other directory
2. Accept (Completable) argument

Please tell me if it works for you, especially the move to other directory

**Comment by tinmarino** [comment]
Closed by 1f77e47

**Comment by BehindTheBrain** [comment]
So I'm checking out rename functionality currently, and I notice 2 things about this issue:

1. From cursory exploration, it seems that this feature is working on native syntax on dev, with the exception that it only works when doing
`:VimwikiRenameFile newpath/newname`
but not
`leader wr` then newpath/newname, 
which still has blocks against any path in the new name.

Unless I'm missing something, this issue being addressed means we don't need the following guard anymore?
```patch
diff --git a/autoload/vimwiki/base.vim b/autoload/vimwiki/base.vim
index 1917718..1d4872e 100644
--- a/autoload/vimwiki/base.vim
+++ b/autoload/vimwiki/base.vim
@@ -1760,10 +1760,6 @@ function! s:input_rename_link() abort
   let new_link = input('Enter new name: ')
 
   " Guard: Check link
-  if new_link =~# '[/\\]'
-    call vimwiki#u#error('Cannot rename to a filename with path!')
-    return
-  endif
   if substitute(new_link, '\s', '', 'g') ==? ''
     call vimwiki#u#error('Cannot rename to an empty filename!')
     return
```

2. There is no documentation of this feature added?

Should this issue be re-opened? A new one created?
